src/cosmos-copilot.WebApp/Program.cs

```diff
@@ -45,41 +88,8 @@ public static void RegisterConfiguration(this WebApplicationBuilder builder)
 
     public static void RegisterServices(this IServiceCollection services)
     {
-        services.AddSingleton<CosmosDbService, CosmosDbService>((provider) =>
-        {
-            var cosmosDbOptions = provider.GetRequiredService<IOptions<CosmosDb>>();
-            if (cosmosDbOptions is null)
-            {
-                throw new ArgumentException($"{nameof(IOptions<CosmosDb>)} was not resolved through dependency injection.");
-            }
-            else
-            {
-                return new CosmosDbService(
-                    endpoint: cosmosDbOptions.Value?.Endpoint ?? String.Empty,
-                    databaseName: cosmosDbOptions.Value?.Database ?? String.Empty,
-                    chatContainerName: cosmosDbOptions.Value?.ChatContainer ?? String.Empty,
-                    cacheContainerName: cosmosDbOptions.Value?.CacheContainer ?? String.Empty,
-                    productContainerName: cosmosDbOptions.Value?.ProductContainer ?? String.Empty,
-                    productDataSourceURI: cosmosDbOptions.Value?.ProductDataSourceURI ?? String.Empty
-                );
-            }
-        });
-        services.AddSingleton<OpenAiService, OpenAiService>((provider) =>
-        {
-            var openAiOptions = provider.GetRequiredService<IOptions<OpenAi>>();
-            if (openAiOptions is null)
-            {
-                throw new ArgumentException($"{nameof(IOptions<OpenAi>)} was not resolved through dependency injection.");
-            }
-            else
-            {
-                return new OpenAiService(
-                    endpoint: openAiOptions.Value?.Endpoint ?? String.Empty,
-                    completionDeploymentName: openAiOptions.Value?.CompletionDeploymentName ?? String.Empty,
-                    embeddingDeploymentName: openAiOptions.Value?.EmbeddingDeploymentName ?? String.Empty
-                );
-            }
-        });
+        services.AddSingleton<CosmosDbService, CosmosDbService>();
+        services.AddSingleton<OpenAiService, OpenAiService>();
```

Comment:
Change this line
 var semanticKernalOptions = provider.GetRequiredService<IOptions<SemanticKernel>>();

to this.
var semanticKernalOptions = provider.GetRequiredService<IOptions<OpenAi>>();

can also get rid loading the configuration for SemanticKernel completely in RegisterConfiguration or maybe keep around and move configurations from Chat into it. That might make more sense depending on what Aayush does with the Semantic Kernel service class.